src/cpp/desktop/DesktopMain.cpp

```diff
@@ -246,6 +246,22 @@ int main(int argc, char* argv[])
       if (error)
          LOG_ERROR(error);
 
+      // attempt to remove stale lockfiles (these can cause Desktop launch
+      // to fail on Windows, and it's much more likely that an encountered
+      // lockfile was left behind by an earlier crash rather than a
+      // race between two RStudio instances)
```

Comment:
How about only cleaning up the lockfile if it's older than some short threshold (say, 10 seconds)? (seems like that would allow crash recovery while still preserving some/most of the race condition protection)

Reply:
Good idea!